### PR TITLE
Python 3.5 final is available.

### PR DESCRIPTION
**Pull request by nedbat**

### CODE REVIEW

File: appveyor/install.ps1 [hunk 1]
```diff
@@ -162,7 +162,8 @@ function InstallPip ($python_home) {
         $webclient = New-Object System.Net.WebClient
         $webclient.DownloadFile($GET_PIP_URL, $GET_PIP_PATH)
         Write-Host "Executing:" $python_path $GET_PIP_PATH
-        Start-Process -FilePath "$python_path" -ArgumentList "$GET_PIP_PATH" -Wait -Passthru
+        Get-ChildItem "$GET_PIP_PATH" -force
```
Comment (ogrisel) [inline appveyor/install.ps1]: Why is this needed?  What does it do?

Comment (ogrisel) [inline appveyor/install.ps1]: Ah ok it's a debug for the previous commit failure. sorry for the noise.

Comment (nedbat) [inline appveyor/install.ps1]: Yeah, sorry, the pip failures are going to require some awkward frequent-commit debugging...  I guess I can do that in coverage.py repo actually....

**Comment by nedbat** [comment]
BTW: I changed the get-pip.py invocation so that we can see its output.  It was failing for a distant reason: https://github.com/ThomasWaldmann/argparse/issues/91   But now we'll be able to see more information from it in any case.

**Comment by ogrisel** [comment]
Now  that the pip / argparse thing is fixed, we get a real compiler issue with Python 3.5:

```
%CMD_IN_ENV% python setup.py nosetests
...
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Python35\include -IC:\Python35\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\wdf\ucrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\wdf\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\wdf\um" "-IC:\Program Files (x86)\Windows Kits\10\include\wdf\winrt" /Tcpyappveyordemo/extension.c /Fobuild\temp.win32-3.5\Release\pyappveyordemo/extension.obj
extension.c
C:\Python35\include\pyconfig.h(68): fatal error C1083: Cannot open include file: 'io.h': No such file or directory 
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\cl.exe' failed with exit status 2
Command exited with code 1
```

https://ci.appveyor.com/project/ogrisel/python-appveyor-demo/build/1.0.176/job/tgdhpcpw8jp3jpb5

So the `CMD_IN_ENV` script is probably not handling properly the final release of 3.5.0 for some reason.

**Comment by ogrisel** [comment]
Actually the compiler issue is probably not related to our  `CMD_IN_ENV` but maybe from the fact that the handling of the `vcruntime140.dll` by distutils has changed between 3.5.0rc4 and final:

http://stevedower.id.au/blog/building-for-python-3-5-part-two/

However I don't understand how this is related to the lookup of the `io.h` header file. Maybe this is caused by the fact that several versions of VS are installed concurrently on the appveyor hosts?

**Comment by ogrisel** [comment]
I googled a bit but found nothing convincing. @zooba do you have the list of `-I` flags on a windows building that works on your side? That should help us understand what's wrong in our setup. Where is the `io.h` supposed to come from?

**Comment by ionelmc** [comment]
@nedbat @ogrisel the io.h failure is caused by a bug in WDK (it tramples the ucrt cause there's a sorting bug in one of their damn batch scripts). See this commit for fix/details: https://github.com/ionelmc/python-lazy-object-proxy/commit/ca5e65c35db8fa9b65fc64d0df51a743d831d657#diff-d7c025352b794932ee322a013be90c90R22

**Comment by ogrisel** [comment]
Thanks very much @ionelmc this is really helpful. I won't have time to include this workaround in `python-appveyor-demo` this WE so if someone else want to do it first please feel free to do so.

Otherwise I might work on that next week.

**Comment by nedbat** [comment]
I applied @ionelmc's fix.  Looks good.

**Comment by ogrisel** [comment]
Thanks all, this is great. Merging!
